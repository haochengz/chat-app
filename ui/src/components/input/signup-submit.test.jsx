
import React from 'react'
import { shallow } from 'enzyme'

import SignupSubmit from './signup-submit'

/*
<SignupSubmit
  stage={this.state.submitBtnName}
  submit={this.whenSubmit.bind(this)}
  query={this.whenQuery.bind(this)}
  done={!this.done()}
  registrant={{
    username: this.state.username.v,
    email: this.state.email.v,
    password: this.state.password.v,
    identity: this.state.identity.v
  }}
/>
*/

describe('<SignupSubmit stage={\'SIGN UP\'}/>', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SignupSubmit stage={'SIGN UP'}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should calling props.query with ok if given username and email is valid for registration', () => {
    const util = require('../../util/user-util')
    util.queryExistenceOnServer = jest.fn(() => Promise.resolve('ok'))
    const whenQuery = (status, payload) => {
      expect(status).toBe('ok')
    }
    const wrapper = shallow(
      <SignupSubmit
        stage={'SIGN UP'}
        query={whenQuery}
        registrant={{
          username: 'testuser',
          email: 'testuser@gmail.com'
        }}
      />
    )
    wrapper.simulate('click')
    expect(util.queryExistenceOnServer).toBeCalled()
  })

  it('should calling props.query with error if given username or email is invalid for registration', () => {
    const util = require('../../util/user-util')
    util.queryExistenceOnServer = jest.fn(() => Promise.reject('username'))
    const whenQuery = (status, payload) => {
      expect(status).toBe('taken', {
        btn: 'SIGN UP',
        hasError: true,
        error: 'Username already been taken',
        originError: 'username'
      })
    }
    const wrapper = shallow(
      <SignupSubmit
        stage={'SIGN UP'}
        query={whenQuery}
        registrant={{
          username: 'testuser',
          email: 'testuser@gmail.com'
        }}
      />
    )
    wrapper.simulate('click')
    expect(util.queryExistenceOnServer).toBeCalled()
  })
})

describe('<SignupSubmit stage={\'CONTINUE\'}/>', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SignupSubmit stage={'CONTINUE'}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should calling props.submit with ok if registration were finish', () => {
    const util = require('../../util/user-util')
    util.signUpAs = jest.fn(() => Promise.resolve())
    const whenSubmit = (status, payload) => {
      expect(status).toBe('ok')
    }
    const wrapper = shallow(
      <SignupSubmit
        stage={'SUBMIT'}
        submit={whenSubmit}
        registrant={{
          username: 'testuser',
          email: 'testuser@gmail.com'
        }}
      />
    )
    wrapper.simulate('click')
    expect(util.signUpAs).toBeCalled()
  })

  it('should calling props.submit with error if registration failed', () => {
    const util = require('../../util/user-util')
    util.signUpAs = jest.fn(() => Promise.reject())
    const whenSubmit = (status, payload) => {
      expect(status).toBe('error')
    }
    const wrapper = shallow(
      <SignupSubmit
        stage={'SUBMIT'}
        submit={whenSubmit}
        registrant={{
          username: 'testuser',
          email: 'testuser@gmail.com'
        }}
      />
    )
    wrapper.simulate('click')
    expect(util.signUpAs).toBeCalled()
  })
})