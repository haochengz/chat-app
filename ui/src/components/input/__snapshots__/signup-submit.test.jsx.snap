// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupSubmit stage={'CONTINUE'}/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignupSubmit
    stage="CONTINUE"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeStyle": Object {},
      "children": "CONTINUE",
      "disabled": false,
      "inline": false,
      "loading": false,
      "onClick": [Function],
      "prefixCls": "am-button",
      "size": "large",
      "type": "primary",
    },
    "ref": null,
    "rendered": "CONTINUE",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeStyle": Object {},
        "children": "CONTINUE",
        "disabled": false,
        "inline": false,
        "loading": false,
        "onClick": [Function],
        "prefixCls": "am-button",
        "size": "large",
        "type": "primary",
      },
      "ref": null,
      "rendered": "CONTINUE",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<SignupSubmit stage={'SIGN UP'}/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignupSubmit
    stage="SIGN UP"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeStyle": Object {},
      "children": "SIGN UP",
      "disabled": false,
      "inline": false,
      "loading": false,
      "onClick": [Function],
      "prefixCls": "am-button",
      "size": "large",
      "type": "primary",
    },
    "ref": null,
    "rendered": "SIGN UP",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeStyle": Object {},
        "children": "SIGN UP",
        "disabled": false,
        "inline": false,
        "loading": false,
        "onClick": [Function],
        "prefixCls": "am-button",
        "size": "large",
        "type": "primary",
      },
      "ref": null,
      "rendered": "SIGN UP",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
